draft-irtf-cfrg-hpke.md

```diff
@@ -296,6 +299,89 @@ plaintexts.
 The procedures described in this session are laid out in a
 Python-like pseudocode.  The ciphersuite in use is left implicit.
 
+## Key Schedule
+
+The variants of HPKE defined in this document share a common
+mechanism for translating the inputs to the protocol into an
+encryption context.  The inputs to the key schedule are as follows:
+
+* `pkR` - The receiver's public key
+* `zz` - A shared secret generated via the KEM for this transaction
+* `enc` - An encapsulated key produced by the KEM for the receiver
+* `info` (optional) - Application-supplied information
+* `psk` (optional) - A pre-shared secret held by both the initiator
+  and the receiver.
+* `pskID` (optional) - An identifier for the PSK
+* `pkI` (optional) - The initiator's public key
+
+If any of the optional values is not present (e.g., set to `None`
+in a Python implementation), then it is treated in the same way as
+an empty octet string.
+
+The key and nonce produced by this algorithm have the property that
+they are only known to the holder of the receipient private key, and
+the party that ran the KEM to generate `zz` and `enc`.  If the `psk`
+and `pskID` arguments are provided, then the recipient is assured
+that the initiator held the PSK.  If the `pkIm` argument is
+provided, then the recipient is assued that the initator held the
+corresponding private key (assuming that `zz` and `enc` were
+generated using the AuthEncap / AuthDecap methods; see below).
+
+~~~~~
+default_pkIm = zero(Npk)
+default_salt = zero(Nh)
+default_pskId = zero(0)
+
+def VerifyMode(mode, salt, pskID, pkIm):
+  got_psk = (salt != default_salt and pskID != default_pskID)
```

Comment:
nit: can we rename `salt` to `psk`? I had to go down to see its invocation to learn that salt is really the psk here.

Reply:
(I recognize we use `salt` since it's more natural for `Extract`, though `psk` I think being clear about the nature of the value is probably better here.